stravalib/model.py

```diff
@@ -11,11 +11,20 @@
 from stravalib import exc
 from stravalib import unithelper as uh
 
-from stravalib.attributes import (META, SUMMARY, DETAILED, Attribute,
-                                  TimestampAttribute, LocationAttribute,
-                                  EntityCollection, EntityAttribute,
-                                  TimeIntervalAttribute, TimezoneAttribute,
-                                  DateAttribute, ChoicesAttribute)
+from stravalib.attributes import (
+    META,
+    SUMMARY,
+    DETAILED,
+    Attribute,
```

Comment:
we have a class called `Attribute`. because attribute is a term for an element  in a Python class it confused me. initially i thought incorrectly that it was an actual attribute called in a way that i didn't understand but afterin talking with friends i am clear it's a class. it's a class called within another class over and over. 

Should we consider naming conventions that do not use existing Python terms?

Reply:
Are you familiar with Python's descriptor protocol? All these `Attribute` objects within model objects are descriptors.

I agree that the naming may be confusing. Whenever we change to Pydantic (and generated model classes), all this logic within the `attributes` module will go (at least all the (un-) marshaling and typing).

Reply:
thank you @jsamoocha i think i need to learn a bit moer about descriptor protocols. I first looked at the code quickly and thought perhaps it was an actual attribute. but then as the docs were rendering i realized (with input from a friend) that it was actually a class called Attribute. but it sounds like i need to read about descriptor protocols.